From 20519db38364c567f041a53b09f885d0b768946f Mon Sep 17 00:00:00 2001
From: Bhargav Upperla <bhargavuln@codeaurora.org>
Date: Tue, 1 May 2018 12:29:10 -0700
Subject: [PATCH 06/14] Add touch boost override

Similar to 1.3Ghz kernel touch boost.
Automated tests inject input events directly
to the android input interface via 'input tap' etc.

Boost that a user will normally experience will
not trigger since automation bypasses the input
driver framework.

Add similar boost in powerHAL layer to
help boost android-level injected input or
INTERACTION events.

Change-Id: I948e22afddb9cc9dd6e0ee203938b325a6f6d15e
---
 power-8150.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/power-8150.c b/power-8150.c
index 0ca65ae..6cc4ed4 100644
--- a/power-8150.c
+++ b/power-8150.c
@@ -92,3 +92,21 @@ int set_interactive_override(struct power_module *module, int on)
     return HINT_HANDLED;
 }
 
+void interaction(int duration, int num_args, int opt_list[]);
+
+int power_hint_override(struct power_module *module, power_hint_t hint, void *data)
+{
+    int ret_val = HINT_NONE;
+    switch(hint) {
+        case POWER_HINT_INTERACTION:
+        {
+            int resources[] = {0x40800100, 0x514};
+            int duration = 100;
+            interaction(duration, sizeof(resources)/sizeof(resources[0]), resources);
+            ret_val = HINT_HANDLED;
+        }
+        default:
+            break;
+    }
+    return ret_val;
+}
-- 
2.17.1

